prov/efa/src/efa_base_ep.c

```diff
@@ -354,8 +354,12 @@ int efa_base_ep_getname(fid_t fid, void *addr, size_t *addrlen)
 #if HAVE_EFA_DATA_IN_ORDER_ALIGNED_128_BYTES
 bool efa_base_ep_support_op_in_order_aligned_128_bytes(struct efa_base_ep *base_ep, enum ibv_wr_opcode op)
 {
-	return ibv_query_qp_data_in_order(base_ep->qp, op,
-					  IBV_QUERY_QP_DATA_IN_ORDER_ALIGNED_128_BYTES);
+	int caps;
+
+	caps = ibv_query_qp_data_in_order(base_ep->qp, op,
+					  IBV_QUERY_QP_DATA_IN_ORDER_RETURN_CAPS);
+
+	return caps == IBV_QUERY_QP_DATA_IN_ORDER_ALIGNED_128_BYTES;
```

Comment:
I would do something like:
return !!(caps & IBV_QUERY_QP_DATA_IN_ORDER_ALIGNED_128_BYTES)

Reply:
The proposed change makes sense.  Note that this PR and an equivalent patch has been merged into v1.18.x branch.  Any update would need to target both main and v1.18.x.